lib/routes/token.js

```diff
@@ -89,6 +101,60 @@ function confirmRefreshToken(params) {
   });
 }
 
+function confirmJwt(params) {
+  var assertion = params.assertion;
+  logger.debug('jwt.confirm', assertion);
+
+  return JWTOOL.verify(assertion).catch(function(err) {
+    logger.info('jwt.invalid.verify', err.message);
+    throw AppError.invalidAssertion();
+  }).then(function(payload) {
+    logger.verbose('jwt.payload', payload);
+
+    var client = SERVICE_CLIENTS[payload.iss];
+
+    if (!client) {
+      logger.debug('jwt.client.unknown', payload.iss);
+      throw AppError.unknownClient(payload.iss);
+    }
+
+    var uid = payload.sub;
+    if (!uid || !(uid.length === 32 && HEX_STRING.test(uid))) {
+      logger.debug('jwt.invalid.sub', uid);
+      throw AppError.invalidAssertion();
+    }
```

Comment:
I don't think 1 is really superior to the other. I think I proposed that because I was looking at the existing `browserid` module used with fxa's browserid assertions, and then proceeded to forget about when writing the code.